gateway/mw_auth_key.go

```diff
@@ -107,22 +108,31 @@ func (k *AuthKey) ProcessRequest(_ http.ResponseWriter, r *http.Request, _ inter
 	if !keyExists {
 		// fallback to search by cert
 		session, keyExists = k.CheckSessionAndIdentityForValidKey(certHash, r)
-		certHash = session.KeyID
```

Comment:
this override is not necessary here otherwise in case of mTLS it will be always diff than `session.Certificate`